### ISSUE TITLE
Please add `proxies=proxies` capability to any functions call to `requests`

### ISSUE THREAD

**Issue by cregouby** [deepset-ai/FARM#115]
**Is your feature request related to a problem or particular use case?**
As a corporate user, I need to pass through proxies for any https request. Currently, framework initialization commands are missing the `proxies=proxies` parameter pass-trough to the `requests` library, making initalization very painful with manual download and badly reproducible.

**Describe the solution you'd like**
Please incorporate `proxies=proxies` parameter to any call to the `requests` library so that initialisation like
```
proxies = {'https': config['https_proxy']}
tokenizer = BertTokenizer(
    vocab_file="bert-base-cased",
    proxies = proxies,
    do_lower_case=False)
```
is now OK through corporate proxies.

**Additional context**
I manage to workaround it through merging with [huggingface/transformers/.../file_utils.py](https://github.com/huggingface/transformers/blob/master/transformers/file_utils.py) web related functions...

**Comment by Timoeller** [comment]
Can it be that this functionality is already in place? Can you double check @tholor please.

Did you, @cregouby  try to use the BertTokenizer like this? Does it produce any errors?

**Comment by cregouby** [comment]
Yes, from the tutorial notebook, right now executing
```
tokenizer = BertTokenizer.from_pretrained(
    pretrained_model_name_or_path="bert-base-cased",
    do_lower_case=False)
```
gives
```
10/15/2019 09:11:52 - INFO - pytorch_transformers.file_utils -   https://s3.amazonaws.com/models.huggingface.co/bert/bert-base-cased-vocab.txt not found in cache, downloading to /tmp/tmpg0u0g9n0
10/15/2019 09:11:52 - ERROR - pytorch_transformers.tokenization_utils -   Couldn't reach server to download vocabulary.
```

**Comment by Timoeller** [comment]
How about you adjust the code in our tutorial notebook to:
```
tokenizer = BertTokenizer.from_pretrained(
    pretrained_model_name_or_path="bert-base-cased",
    proxies = proxies,
    do_lower_case=False)
```
and set the proxies yourself?

**Comment by tholor** [comment]
Supporting proxies is definitely an important feature since many corporates face this issue. We will have a look how to integrate it most smoothly into FARM. I see different options from a) passing the proxy arg all the way down to the requests library to b) reading a global environment variable. @tanaysoni can you please follow up on this?

**Comment by tholor** [comment]
Thanks for the update @cregouby and the PR. 
I see 3 parts of the FARM code that rely on proxies:
1) `Tokenizer.load()` => should be okay, since we are just passing the proxy arg to the transformers 
2) `LanguageModel.load()` => should be okay, after your PR #132 
3) [_download_extract_downstream_data()](https://github.com/deepset-ai/FARM/blob/3f10b8282ffa3e4101da5e5595a4eb63413ec813/farm/data_handler/utils.py#L104) (Downloads standard datasets like Squad etc.) => not covered yet, I will probably find some time to work on this next week

**Comment by cregouby** [comment]
you are right, _download_extract_downstream_data() need some rework.

**Comment by tholor** [comment]
Fixed by #136
